hub/config/src/main/java/uk/gov/ida/hub/config/exceptions/ExceptionFactory.java

```diff
@@ -26,6 +26,12 @@ public NotFoundException createNoDataForEntityException(String entityId) {
         LOG.error(message);
         return new NotFoundException(message);
     }
+    
+    public NotFoundException createNoTranslationForLocaleException(String locale) {
```

Comment:
Thanks for the feedback. `204` is generally used to indicate that there is no content to include in the response (e.g. could be used as a response to a successful `PUT` request), but doesn't signify that the content doesn't exist. This exception is for the scenario where the resource (or rather translation) corresponding to the request simply doesn't exist, and therefore `404` is used.

Regarding creating a custom exception type, it's a good point but I'm not sure if it will add additional benefit at this point in time. It would have made a lot more sense if we had code flow where it was possible to throw the same exception for different resources, at which point it would have been useful to be able to check exception type. We could certainly upgrade it to a custom exception type in the future when we need it, but currently feels like pre-mature optimisation to me. Feel free to correct me if I'm missing something.